Project/combat.py

```diff
@@ -0,0 +1,69 @@
+import random
+from player import Player
+
+class Combat:
+    """Handles turn-based combat mechanics."""
+    
+    def __init__(self, player):
+        self.player = player
+        self.enemy = self.generate_enemy()
+
+    def generate_enemy(self):
+        """Dynamically generates an enemy with randomized stats."""
+        enemies = [
```

Comment:
I would move all the enemy data to the top as a constant for consistency if you don't want to move on with creating a json file for those data.

```suggestion
ENEMIES = [
    {"name": "Goblin", "health": 50, "attack": 10, "defense": 2, "evasion": 5},
    {"name": "Orc", "health": 80, "attack": 15, "defense": 4, "evasion": 3},
    {"name": "Dark Mage", "health": 60, "attack": 12, "defense": 3, "evasion": 7}
]

def generate_enemy(self):
    return random.choice(ENEMIES).copy()

```